### DISCUSSION TITLE
update directus_files metadata from API

### DISCUSSION THREAD

**Discussion by mmmaxxxe** [directus/directus#16405] [Archive]
Hi there!

i have to update metadata on directus_files via API:

1) if i use /files/:id i receive an error 

"errors":[{"message":"No files where included in the body","extensions":{"code":"INVALID_PAYLOAD"}}]}

2) if i use /directus_files/:id i receive another error:

{"errors":[{"message":"Route \/directus_files\/xxxxxx doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}

What is the best strategy to accomplish this?

Thanks

**Comment by azrikahar** [comment]
> if i use /files/:id i receive an error
"errors":[{"message":"No files where included in the body","extensions":{"code":"INVALID_PAYLOAD"}}]}

Reference: https://docs.directus.io/reference/files.html#update-a-file

If you are only updating the metadata and not the file, make sure _not_ to use `multipart/data` request. You should use `application/json` instead. The error you are encountering seems to indicate you were using `multipart/data`, which means this update needs to include a new file as the new replacement, so it is (correctly) throwing error saying file not included.

> if i use /directus_files/:id i receive another error:
{"errors":[{"message":"Route /directus_files/xxxxxx doesn't exist.","extensions":{"code":"ROUTE_NOT_FOUND"}}]}

This error is correct as system collections does not need the `directus_` prefix.
